c3: update get started guides with info on deploying with git repos. (#12966)
cloudflare/cloudflare-docs: content/workers/get-started/guide.md

4. `package-lock.json`: Refer to [`npm` documentation on `package-lock.json`](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json).
5. `node_modules`: Refer to [`npm` documentation `node_modules`](https://docs.npmjs.com/cli/v7/configuring-npm/folders#node-modules).

### Create a new Worker Project from an external source

> In addition to creating new projects from its own templates, C3 also supports creating new projects from git repositories. In order to do that, you can open your terminal window and run:

Reviewer: ```suggestion
In addition to creating new projects from C3 templates, C3 also supports creating new projects from Git repositories. To create a new project from a Git repository, open your terminal and run:
```

> {{<tabs labels="npx | yarn | pnpm | bun">}}

Reviewer: ```suggestion
{{<tabs labels="npm | yarn | pnpm | bun">}}
```

{{<tab label="npm" default="true">}}
```sh

> $ npx create-cloudflare@latest --template [<SOURCE>]

Reviewer: Same comment about standardizing `npm` here
